turb/public/turb.f90

```diff
@@ -130,14 +131,23 @@ subroutine calc_frg24_w(pr, tau, r0, hb, db, ks, n, w, withTC, ierr, lamhat, l2h
      if (present(lamhat)) then
         ! lamhat and l2hat already calculated, so can pass as arguments
         if(withTC) then
-           w = wf_withTC(pr, tau, r0, hb, db, ks, n, .false., lamhat, l2hat)
+           if (safety == 0) then
+             ! TODO: is it bad for this minimization to be occurring here AND in thermohaline.f90?
```

Comment:
I think this is fine. This `calc_frg24_w ` routine just exists to give a public interface for calling the velocities so we can use it in turb/plotter. Any actual stellar evolution goes through the `get_D_thermohaline` routine that is calling the stuff in `thermohaline.f90` independent of this, because right now mesa/star just wants a thermohaline composition diffusion coefficient.

Reply:
But this structure does end up causing a lot of duplicated code at this point. It would be better to have all that code live in a velocity routine in `private/thermohaline.f90`, and leave this public interface as just a wrapper to call that private routine.

Reply:
Rich and I are talking about this now and are putting a pin in this to resolve later.